src/main/kotlin/uk/gov/gdx/datashare/services/EventDataService.kt

```diff
@@ -119,20 +120,25 @@ class EventDataService(
     val enrichmentFields =
       acquirerSubscriptionEnrichmentFieldRepository.findAllByAcquirerSubscriptionId(subscription.acquirerSubscriptionId)
         .map { it.enrichmentField }
+
     return EventNotification(
       eventId = event.id,
       eventType = subscription.eventType,
       sourceId = event.dataId,
       dataIncluded = if (!callbackEvent) includeData else null,
-      enrichmentFields = if (!callbackEvent) enrichmentFields else null,
-      eventData = if (includeData) {
-        deathNotificationService.getEnrichedPayload(
-          event.dataId,
-          enrichmentFields,
-        )
-      } else {
-        null
-      },
+      enrichmentFields = enrichmentFields,
+      eventData = if (includeData) callbackAndEnrichData(subscription, event, enrichmentFields) else null,
     )
   }
+
+  @Suppress("IMPLICIT_CAST_TO_ANY")
```

Comment:
any reason for this vs

```
: Any?
```

on the function return?

Reply:
No! - Will change